libgraphics/src/legacygradients.cpp

```diff
@@ -339,7 +340,7 @@ class MCGGeneralizedGradientShader : public SkShader
             {
                 if ((int)m_buffer_width < count * GRADIENT_AA_SCALE)
                 {
-                    uindex_t t_size = m_buffer_width * GRADIENT_AA_SCALE * GRADIENT_AA_SCALE;
+                    uindex_t t_size = m_buffer_width * GRADIENT_AA_SCALE;
                     if (!MCMemoryResizeArray(count * GRADIENT_AA_SCALE * GRADIENT_AA_SCALE, m_buffer, t_size))
```

Comment:
Hmmm - So ResizeArray is meant to work to update an existin garray ptr and size - so could this not be:
```
MCMemoryResizeArray(count * GRADIENT_AA_SCALE, m_buffer, m_buffer_size)
```

Reply:
@runrevmark m_buffer_width is specified as pixel_span_width * GRADIENT_AA_SCALE and the full buffer size as pixel_span_width * GRADIENT_AA_SCALE^2 so it does need to be count * GRADIENT_AA_SCALE * GRADIENT_AA_SCALE

Reply:
Got it - m_buffer_width is old-count * AA_SCALE :)